src/SFA.DAS.AdminService.Web/Controllers/Apply/ApplicationController.cs

```diff
@@ -258,6 +259,142 @@ public async Task<IActionResult> Assessment(Guid applicationId, int sequenceNo,
+                }
+
+                sequenceVm.IncrementCurrentVersionIndex();
+                if(sequenceVm.CurrentVersionIndex.HasValue)
+                {
+                    return RedirectToAction(nameof(WithdrawalDateCheck), new { currentVersionIndex = sequenceVm.CurrentVersionIndex.Value });
+                }
+            }
+
+            return RedirectToAction(nameof(Assessment));
+        }
+
+        [HttpPost("/Applications/{applicationId}/{backAction}/{backController}/Sequence/{sequenceNo}/WithdrawalDateChange/{backOrganisationId?}")]
+        public async Task<IActionResult> WithdrawalDateChange(Guid applicationId, int sequenceNo, BackViewModel backViewModel, string effectiveToDay, string effectiveToMonth, string effectiveToYear, int currentVersionIndex)
+        {
+            var application = await _applyApiClient.GetApplication(applicationId);
+            var organisation = await _apiClient.GetOrganisation(application.OrganisationId);
+
+            var activeApplySequence = application.ApplyData.Sequences.Where(seq => seq.IsActive && !seq.NotRequired).OrderBy(seq => seq.SequenceNo).FirstOrDefault();
+
+            var sequence = await _qnaApiClient.GetSequence(application.ApplicationId, activeApplySequence.SequenceId);
+            var sections = await _qnaApiClient.GetSections(application.ApplicationId, sequence.Id);
+
+            var sequenceVm = new WithdrawalDateCheckViewModel(application, organisation, sequence, sections,
+                activeApplySequence.Sections,
+                backViewModel.BackAction,
+                backViewModel.BackController,
+                backViewModel.BackOrganisationId,
+                currentVersionIndex);
+
+            var errorMessages = new Dictionary<string, string>();
+
+            string effectiveToDateText = $"{effectiveToDay}/{effectiveToMonth}/{effectiveToYear}";
+            if (!DateTime.TryParse(effectiveToDateText, out DateTime effectiveToDate))
+            {
+                errorMessages["RequestedWithdrawalDate"] = "Please enter a valid date.";
+            }
+
+            if (errorMessages.Any())
+            {
+                foreach (var error in errorMessages)
+                {
+                    ModelState.AddModelError(error.Key, error.Value);
+                }
+
+                return View(nameof(WithdrawalDateChange), sequenceVm);
+            }
+
+            if (sequenceVm.SequenceNo == ApplyConst.ORGANISATION_WITHDRAWAL_SEQUENCE_NO)
+            {
+                await UpdateOrganisationStandardWithdrawalDate(organisation.EndPointAssessorOrganisationId, null, null, effectiveToDate);
+            }
+            else
+            {
+                await UpdateOrganisationStandardWithdrawalDate(organisation.EndPointAssessorOrganisationId, sequenceVm.StandardReference, sequenceVm.Versions[sequenceVm.CurrentVersionIndex.Value], effectiveToDate);
```

Comment:
Got a null reference exception here while trying to approve an application for a standard which only had one version, the versions property on the model was null. Worth a double check

This was withdrawing from the brewer standard, and modifying the withdrawal date page